packages/extension-utils/src/OAuthForm/src/components/OAuthForm.tsx

```diff
@@ -113,88 +105,31 @@ export const OAuthForm = ({
     )
   }
 
-  const saveConfig = (baseUrl: string, webUrl: string) => {
-    const data = {
-      base_url: baseUrl,
-      looker_url: webUrl,
+  const handleSave = () => {
+    handleSaveConfigAction({
+      configKey,
+      setHasConfig,
       client_id: clientId,
       redirect_uri,
-    }
-    localStorage.setItem(configKey, JSON.stringify(data))
-    saveNewConfigAction(data)
-    if (setHasConfig) setHasConfig(true)
-  }
-
-  const verifyUrl = async (): Promise<ILookerVersions | undefined> => {
-    updateMessageBarAction({ intent: messageBar.intent, text: '' })
-    const versionsUrl = `${apiServerUrl}/versions`
-    try {
-      setFetchedUrlAction(apiServerUrl)
-
-      const versions = await getVersions(versionsUrl)
-
-      if (versions) {
-        updateMessageBarAction({
-          intent: 'positive',
-          text: `Configuration is valid`,
-        })
-        setWebUrlAction(versions.web_server_url)
-      }
-
-      return versions
-    } catch (e: any) {
-      verifyErrorAction(e.message)
-      return undefined
-    }
-  }
-
-  const handleSave = async () => {
-    const versions = await verifyUrl()
-    if (versions) {
-      saveConfig(versions.api_server_url, versions.web_server_url)
-    }
+    })
   }
 
   const handleVerify = () => {
-    verifyUrl()
+    handleVerifyAction()
   }
 
   const handleClear = async (_e: BaseSyntheticEvent) => {
-    localStorage.removeItem(configKey)
-    clearFormAction()
-
-    if (setHasConfig) setHasConfig(false)
-    if (isAuthenticated()) {
-      updateMessageBarAction({
-        intent: 'warn',
-        text: `Please reload the browser page to log out`,
-      })
-    }
+    clearFormAction({
+      configKey,
+      setHasConfig: setHasConfig,
+      isAuthenticated: isAuthenticated(),
+    })
   }
 
   const handleUrlChange = (event: FormEvent<HTMLInputElement>) => {
-    const name = event.currentTarget.name
-    const value = event.currentTarget.value
-    let newApiServerUrl = value
-
-    const newValidationMessages = { ...validationMessages }
-
-    const url = validateUrl(value)
-    if (url) {
-      delete newValidationMessages[name]
-      // Update URL if it's been cleaned up
-      newApiServerUrl = url
-    } else {
-      newValidationMessages[name] = {
-        message: `'${value}' is not a valid url`,
-        type: 'error',
-      }
-    }
-
-    updateApiServerUrlAction({
-      apiServerUrl: newApiServerUrl,
-      webUrl: '',
-      validationMessages: newValidationMessages as ValidationMessages,
+    handleUrlChangeAction({
```

Comment:
Is this something I could do inline the JSX, or is it better to leave it out of it?

Reply:
you could but I'd recommend against it as it makes debugging harder